### ISSUE TITLE
Sort & Filter Submissions

### ISSUE THREAD

**Issue by r-bartlett-gsa** [GSA/Challenge_platform#57]
### User Story

As a challenge manager, in order to manage the progress of evaluations efficiently, I would like to be able to sort submissions based on submission and evaluation progress data.

**Acceptance criteria:**

A challenge manager can sort and filter the submissions (on the submission list view) by:
- [x]  Filter by: Progress status: completed, in progress, not started, recused
- [x]  Filter by: Eligible for Evaluations (Prescreen status)
- [x]  Filter by: Advancement status
- [x] Sort by: Submission ID: low to high, high to low
- [x] #418
- [x] Sort by: Average score (for completed evaluations only): low to high, high to low

**Dev tasks:**

- [x] #207

### Definition of Done

**Doing (dev team)**
- [x] Code complete 
- [x] Code is organized appropriately
- [x] Any known trade offs are documented in the associated GH issue
- [x] Code is documented, modules, shared functions, etc.
- [x] Automated testing has been added or updated in response to changes in this PR
- [x] The feature is smoke tested to confirm it meets requirements
- [x] Database changes have been peer reviewed for index changes and performance bottlenecks
- [x] PR that changes or adds UI
  - [x] include a screenshot of the WAVE report for the altered pages
  - [x] Confirm changes were validated for mobile responsiveness
- [x] PR approved / Peer reviewed 
- [x] Security scans passed
- [x] Automate accessibility tests passed
- [x] Build process and deployment is automated and repeatable
- [x] Feature toggles if appropriate
- [x] Deploy to staging
- [x] Move card to testing column in the board
---
**Staging**
- [ ] Accessibility tested (Marni)
  - [ ]  Keyboard navigation
  - [ ]  Focus confirmed
  - [x] Color contrast compliance
  - [ ] Screen reader testing 
- [x] Usability testing: mobile and desktop (Tracy or Marni)
- [x] Cross browser testing - UI rendering is performant on below listed devices/browsers (Tracy or Marni)
  - [x] Windows/Chrome
  - [x] Windows/Edge
  - [x] Mac/Chrome
  - [x] Mac/Safari
  - [x] iOS/Safari
- [x] AC review (Renata)
- [x] Deploy to production (production-like environment for eval capability) (dev team)
- [x] Move to production column in the board
---
**Production**
- [ ] User and security documentation has been reviewed for necessary updates (Renata/Tracy/Dev team) 
- [ ] PO / PM approved (Jarah or Renata)
- [ ] AC is met and it works as expected (Jarah or Renata)
- [ ] Move to done column in the board (Jarah or Renata)

**Comment by r-bartlett-gsa** [comment]
**Some button styles, colors and labels may have changed and screenshots below may reflect outdated styles. Please refer to [Figma](https://www.figma.com/design/VfdBFG8ShVnOStdSNYy9fR/Challenge.Gov-Component-Library?node-id=2750-62256&node-type=section&t=6lzkQqYVjlLF4EbT-0) for latest designs.**
Misalignment in styles, colors and labels should not affect the completion of development work. There will be additional user stories towards the end of this epic or MVP to bring uniformity to styles and colors.
If significant difference is noticed in features / functionality, please contact Renata.

![Image](https://github.com/user-attachments/assets/b092f7a1-ce2d-4258-a1fc-11b1b54511a7)

**Comment by stepchud** [comment]
@r-bartlett-gsa for the Evaluation Progress filter, does Completed mean all Evaluators have completed, Not Started means none of the Evaluators have started, and In Progress means it's neither of those?

The recused filter is not in Figma, is that no longer needed? if needed, does that show submissions with any recused status?

**Comment by r-bartlett-gsa** [comment]
> @r-bartlett-gsa for the Evaluation Progress filter, does Completed mean all Evaluators have completed, Not Started means none of the Evaluators have started, and In Progress means it's neither of those?

@stepchud Yes

> The recused filter is not in Figma, is that no longer needed? if needed, does that show submissions with any recused status?

@stepchud I had recused status filter only on one screen, but now all screens that show expanded filter are updated in Figma. Recused filter is needed and, yes, it shows submissions with a any recused status.

**Comment by r-bartlett-gsa** [comment]
@stepchud Testing feedback:

- Not started filter is not working:

![Image](https://github.com/user-attachments/assets/8811cb03-5b22-4796-8b6f-938607c8bebe)

- Recused filter testing still pending

**Comment by stepchud** [comment]
can be re-tested after #418 is resolved

**Comment by stepchud** [comment]
@r-bartlett-gsa #418 is resolved and this can be tested again
